File: examples/python-example/src/utils/logger.py
"""Logger utility - no internal dependencies."""


class Logger:
    def __init__(self, prefix: str):
        self.prefix = prefix

    def info(self, message: str) -> None:
        print(f"[{self.prefix}] INFO: {message}")

    def error(self, message: str) -> None:
        print(f"[{self.prefix}] ERROR: {message}")

    def warn(self, message: str) -> None:
        print(f"[{self.prefix}] WARN: {message}")

File: examples/go-example/utils/logger.go
package utils

import "fmt"

// Logger provides simple logging functionality.
type Logger struct {
	Prefix string
}

// NewLogger creates a new Logger instance.
func NewLogger(prefix string) *Logger {
	return &Logger{Prefix: prefix}
}

// Info logs an informational message.
func (l *Logger) Info(message string) {
	fmt.Printf("[%s] INFO: %s\n", l.Prefix, message)
}

// Error logs an error message.
func (l *Logger) Error(message string) {
	fmt.Printf("[%s] ERROR: %s\n", l.Prefix, message)
}
